src/test/java/io/jenkins/plugins/appcenter/validator/PathToAppValidatorTest.java

```diff
@@ -61,4 +61,16 @@ public void should_ReturnTrue_When_PathIsRelative_UnixLike() {
         // Then
         assertThat(result).isTrue();
     }
+
+    @Test
+    public void should_ReturnTrue_When_PathContainsEnvVars() {
+        // Given
+        final String value = "path/to/app-${BUILD_NUMBER}.apk";
```

Comment:
The validator simply checks is the path is relative. Whilst it works for this input, I think in general it might give false positives. For example what about:

```
// ${PATH_TO_APP} = /home/jenkins/absolute/path/to/app.apk
final String value = "${PATH_TO_APP}";
```

Which should fail..... but will actually pass.

Reply:
I'm not 100% sure what to do here, to be honest. I like that all the validators do not know anything about Jenkins or the environment it is working under. But in that case the above scenario will occur.

My initial thoughts are to modify `doCheckPathToApp` method in `AppCenterRecorder` to check for an environment variable character (probably `${`) at the start of the input and return a warning instead of an error. We can put some guidance in the help text both in the job configuration and in the warning to ensure users are not using _only_ an environment variable as this may change over time and fail later down the line. Something like this:

```
if (/* looks like it starts with an environment variable*/) {
  return FormValidation.warning(Messages.AppCenterRecorder_DescriptorImpl_warnings_mustNotStartWithEnvVar()); // note: this is generated through the `Messages.properties` file.
}
```

What do you think? If users ignore this warning it will fail later down the line on builds but at least we don't tell them the configuration is valid.

Reply:
Sounds good, will add that and related test

Reply:
@mezpahlan added an additional validator (and related unit tests) and using that in the AppCenterRecorder to trigger warnings.